pkg/controller/networkpolicy/validate.go

```diff
@@ -139,6 +144,28 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(op admv1.Operation, tier s
 	return reason, allowed
 }
 
+// validateRuleName validates if the name of each rule is unique within a policy
+func (v *NetworkPolicyValidator) validateRuleName(ingress, egress []secv1alpha1.Rule) bool {
+	uniqueRuleName := sets.NewString()
+	for _, rule := range ingress {
```

Comment:
```suggestion
	for _, rule := range append(ingress, egress...) {
```
to reduce redundancy.

Reply:
Will this cause additional memory overhead?

Reply:
You must be right from memory perspective. I just thought it's redundant to repeat a piece of code that does same thing for different targets. Personally I feel the overhead of appending short slice (it's just 1 allocation) is minor and worth to reduce code redundancy. But of course there could be another way:
```
uniqueRuleName := sets.NewString()
isUnique := func(s []string) bool {
	for _, i := range s {
		if uniqueRuleName.Has(i) {
			return false
		}
		if i != "" {
			uniqueRuleName.Insert(i)
		}
	}
	return true
}
return isUnique(a) && isUnique(b)
```
The benchmark result (Compare1 is your current way, Compare2 is the append slice way, Compare3 is the function way):
```
BenchmarkCompare1-4      3743316               325 ns/op               0 B/op          0 allocs/op
BenchmarkCompare2-4      2478950               497 ns/op              96 B/op          1 allocs/op
BenchmarkCompare3-4      3480390               327 ns/op               0 B/op          0 allocs/op
```

Reply:
Thx!